department-of-veterans-affairs/vets-website
# 11535 soft launch modal

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

closes https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11535, https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11976, https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11978

## Summary

- Adds soft launch modal to the homepage behind feature flags on staging/prod

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11535
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11975 - issue found during QA here, non-blocking for this PR
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11976 - issue found during QA here, non-blocking for this PR


## Testing done
Local testing with both staging and prod api data.
QA steps:
- watch vets-website with prod api.
- verify that no modal appears on home, or any other mage
- watch vets-website with staging api
- verify that modal renders as expected
- verify functionality of both buttons, and dismissal on close
- verify modal only appears on homepage

## Screenshots
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/61624970/207620182-8198c5b8-d6b3-479f-87fa-da713adb1ea6.png">


## What areas of the site does it impact?
This uses the announcements framework to display a new modal component on the homepage

## Acceptance criteria

- [ ] Meets issue ACs

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
